GVRf/Framework/framework/src/main/java/org/gearvrf/GVRScene.java

```diff
@@ -134,6 +134,7 @@ public void removeAllSceneObjects() {
         head.getParent().removeChildObject(head);
```

Comment:
Yes I think it should. If bindShaders gets called from two different threads it could create a problem. I think the asset loader can do this.